### PR TITLE
*ERROR* the execution returns TypeError: Cannot set property 'transfe…

### PR DESCRIPTION
**Pull request by bgodlin**
…r' of undefined

Initialized the branch for the new entity organisation (https://soramitsu.atlassian.net/browse/SN-868)
Changed graphql schema
Added historyElement handler
Edited transfer handler
Changed from production network to dev
Commented unused code

### CODE REVIEW

File: src/mappings/transfersHandler.ts [hunk 1]
```diff
@@ -1,26 +1,22 @@
...
-    transfer.assetId = assetId.toString();
-    transfer.networkFee = fee;
-    transfer.success = success;
-    transfer.blockHash = event.block.block.hash.toString();
-    transfer.datetime = transferExtrinsic.block.timestamp;
+    const record = await HistoryElement.get(event.extrinsic.extrinsic.hash.toString())
 
-    await transfer.save();
-}
+    record.transfer = {
```
Comment (pgolovkin) [inline src/mappings/transfersHandler.ts:13]: The `Transfer` should be saved before setting it to the `History`.
let transfer = new Transfer(id);
//fill transfer
await transfer.save();

record.transfer = transfer.
await record.save().

File: src/mappings/utils.ts [hunk 2]
```diff
@@ -28,3 +29,15 @@ export const getExtrinsicNetworkFee = (extrinsic: SubstrateExtrinsic): string =>
 export const getExtrinsicId = (extrinsic: SubstrateExtrinsic): string => {
     return `${extrinsic.block.block.hash.toString()}-${extrinsic.idx.toString()}`;
 }
+
+export const ReceiveSwapAmounts = (swapAmount: SwapAmount): string[] => {
```
Comment (pgolovkin) [inline src/mappings/utils.ts:33]: function name should start with lower case `recieveSwapAmounts`

File: project.yaml [hunk 3]
```diff
@@ -557,18 +557,42 @@ network:
...
   - name: main
     kind: substrate/Runtime
-    startBlock: 1000000
+    startBlock: 	32490
     mapping:
       handlers:
-        - handler: handleXYKPools
-          kind: substrate/BlockHandler
+        
+        # - handler: handleXYKPools
```
Comment (pgolovkin) [inline project.yaml:564]: Please, remove this commented code.

File: project.yaml [hunk 4]
```diff
@@ -557,18 +557,42 @@ network:
...
+            module: poolXyk
+            method: withdrawLiquidity
+        
+        - handler: handlerIrohaMigration
+          kind: substrate/CallHandler
+          filter:
+            module: irohaMigration
+            method: migrate
+        
+        # - handler: handleHistoryElement
```
Comment (pgolovkin) [inline project.yaml:597]: How are you going to handler other types of extrinsics?

File: schema.graphql [hunk 5]
```diff
@@ -18,31 +18,45 @@ type Pool @entity {
...
+  assetId: String! 
   amount: String!
-  networkFee: String!
-  success: Boolean!
-  datetime: Date!
 }
+
+type LiquidityOperation @jsonField {
+  type: String!
+  AssetAId: String!
```
Comment (pgolovkin) [inline schema.graphql:53]: Please, use the same naming as for swaps 
  baseAssetId: String!
  targetAssetId: String!
  baseAssetAmount: String!
  targetAssetAmount: String!

File: schema.graphql [hunk 6]
```diff
@@ -18,31 +18,45 @@ type Pool @entity {
...
+  AssetBId: String!
+  AssetAAmount: String!
+  AssetBAmmount: String!
+}
+
+type irohaMigration @jsonField {
+  assetid: String!
```
Comment (pgolovkin) [inline schema.graphql:60]: use came case `assetId`

File: src/mappings/historyElementHandler.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import {SubstrateExtrinsic} from '@subql/types';
+import {HistoryElement} from "../types";
+
+import {checkIfExtrinsicExecuteSuccess, formatU128ToBalance, getExtrinsicId, getExtrinsicNetworkFee} from "./utils";
+
+export async function handleHistoryElement(extrinsic: SubstrateExtrinsic): Promise<void> {
+    
+    logger.warn("New extrinsic")
```
Comment (pgolovkin) [inline src/mappings/historyElementHandler.ts:8]: use `logger.debug` for debugging information.

File: schema.graphql [hunk 8]
```diff
@@ -18,31 +18,45 @@ type Pool @entity {
...
 
-type Transfer @entity {
-  id: ID!
-  extrinsiHash: String!
-  from: String! @index
-  to: String! @index
-  blockHash: String!
-  assetId: String! @index
+type Transfer @jsonField {
+  from: String!
```
Comment (pgolovkin) [inline schema.graphql:45]: add @index

File: schema.graphql [hunk 9]
```diff
@@ -18,31 +18,45 @@ type Pool @entity {
...
-type Transfer @entity {
-  id: ID!
-  extrinsiHash: String!
-  from: String! @index
-  to: String! @index
-  blockHash: String!
-  assetId: String! @index
+type Transfer @jsonField {
+  from: String! 
+  to: String!
```
Comment (pgolovkin) [inline schema.graphql:46]: add @index

File: src/mappings/historyElementHandler.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import {SubstrateExtrinsic} from '@subql/types';
+import {HistoryElement} from "../types";
+
+import {checkIfExtrinsicExecuteSuccess, formatU128ToBalance, getExtrinsicId, getExtrinsicNetworkFee} from "./utils";
+
+export async function handleHistoryElement(extrinsic: SubstrateExtrinsic): Promise<void> {
```
Comment (pgolovkin) [inline src/mappings/historyElementHandler.ts:6]: Seems unused handler.

File: schema.graphql [hunk 11]
```diff
@@ -18,31 +18,45 @@ type Pool @entity {
   updated: Date!
 }
 
-type Swap @entity {
+type HistoryElement @entity {
   id: ID!
-  blockHash: String!
-  extrinsicHash: String!
+  blockHeight: BigInt!
```
Comment (pgolovkin) [inline schema.graphql:23]: it will be good to add here `method` and `module`.

File: project.yaml (hunk 4 above)
Comment (bgodlin) [reply project.yaml:597]: Other extrinsic types are not handled. Once new extrinsics start being used and the need for a new history element type appears, the new handler for a new extrinsic will be added.

File: docker-compose.yml [hunk 12]
```diff
@@ -43,3 +43,4 @@ services:
     command:
       - --name=app
       - --playground
+      - --log-level=debug
```
Comment (pgolovkin) [inline docker-compose.yml:46]: we don't need. When app will be deployed to SubQuery the logger level might be switched in runtime.

File: src/mappings/swapsHandler.ts [hunk 13]
```diff
@@ -1,43 +1,54 @@
...
-    record.address = extrinsic?.extrinsic.signer.toString();
-    record.networkFee = formatU128ToBalance(getExtrinsicNetworkFee(extrinsic));
-    record.selectedMarket = (extrinsic.extrinsic.args[4] as Vec<LiquiditySourceType>).map(lst => lst.toString());
-    record.datetime = extrinsic.block.timestamp;
-    let success = checkIfExtrinsicExecuteSuccess(extrinsic);
-    if (success) {
+    
+    logger.debug("Caught swap extrinsic")
+    
+    const record = new HistoryElement(extrinsic.extrinsic.hash.toString())
```
Comment (pgolovkin) [inline src/mappings/swapsHandler.ts:11]: this block of code is repeated in all handlers. Can it be moved to `utils`? 
I mean the lines from 11th to 19th.

File: src/mappings/irohaMigrationHandler.ts [hunk 14]
```diff
@@ -1,35 +1,35 @@
...
-    
+
     logger.debug("Caught iroha migration extrinsic")
-    
-    const record = new HistoryElement(extrinsic.extrinsic.hash.toString())
 
-    record.blockHeight = extrinsic.block.block.header.number.toBigInt()
+    const record = assignCommonHistoryElemInfo(extrinsic)
+
     record.module = "irohaMigration"
```
Comment (pgolovkin) [inline src/mappings/irohaMigrationHandler.ts:10]: these fields might be filled in `assignCommonHistoryElemInfo` too.
You can use methods: 
`extrinsic.extrinsic.method.method`
`extrinsic.extrinsic.method.section`

**Review by pgolovkin — APPROVED** [review]

**Review by pgolovkin — APPROVED** [review]

**Pull request merged**